### DISCUSSION TITLE
Support pagination data that references a dynamic/templated collection name

### DISCUSSION THREAD

**Discussion by jonsage** [11ty/eleventy#4035] [Enhancement Queue]
I would like to be able to do something like the following in my template for category pages to enable me to use the same template for various categories (_same applies to tags_):

```yaml
pagination:
  data: collections.categories["{{ slug }}"]
# data: collections["{{ tag }}"] 
```

`collections.categories` is a custom collection that picks up all posts with a `category` property in the frontmatter and groups them (see example data below)

`slug` is a property in the frontmatter that forms the url portion of the category page, and the name of the corresponding category collection.

**_includes/layouts/category.njk**
```nunjucks
---
layout: layouts/base.njk
section: category
permalink: /{{ slug }}/{{ pagination.pageNumber }}/
pagination:
  data: collections.categories['{{ slug }}']
  reverse: true
  alias: posts
  size: 2
---
{% for post in posts %}<p>{{ post.title }}</p>{% endfor %}
```

**collections.categories**
```javascript
{
  food: [/*posts*/],
  drink: [/*posts*/]
}
```

**Comment by jonsage** [comment]
Maybe there is a different way to avoid hardcoding collection names throughout when using pagination... seems like it could be a regular occurrence in a average sized site.

**Comment by zachleat** [comment]
Oh hmm—interesting idea! I’ll put this in the enhancement queue!

Tangentially related: https://github.com/11ty/11ty.io/issues/33
See also #335.

**Comment by zachleat** [comment]
This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by danfascia** [comment]
As I read the issues, there are quite a few that are getting at the same issue here but phrased differently because nobody knows how to say it!!

Basically I think a lot of us are confused as to how we can make archive style list pages which take dynamic values (like {{tag}}) and still output a list of items using the native pagination features.

I think that the root of the confusion is the fact that pagination can be _botched_ into generating individual pages...

I think really there should be 2 different things so that pagination can work as expected

There should be:
- Paging - i.e. the output of pages based on a dynamic changing value specified in frontmatter (as above)
- Pagination - exactly as currently implemented, with no changes needed.

This approach would allow pagination of results in a dynamic page because the pagination loop would not have already been used up

**Comment by edwardhorsford** [comment]
@danfascia I think this is what I was getting at in #334.

**Comment by MadeByMike** [comment]
I arrived here and want to add that I think being able to generate an index page for each collection (using pagination or otherwise) seems like a good idea.

**Comment by vhbui02** [comment]
Why closed this? The problem is still there?
